frontend/src/__tests__/cypress/cypress/pages/pipelines/pipelineImportModal.ts

```diff
@@ -1,12 +1,203 @@
+    pipelineImportModal.fillPipelineName(uploadPipelineParams.display_name);
+    pipelineImportModal.fillPipelineDescription(uploadPipelineParams.description);
+    pipelineImportModal.uploadPipelineYaml(pipelineYamlPath);
+    pipelineImportModal.submit();
+
+    // Wait for upload/fetch requests
+    cy.wait('@uploadPipeline').then((interception) => {
+      // Note: contain is used instead of equals as different browser engines will add a different boundary
+      // to the body - the aim is to not limit these tests to working with one specific engine.
+      expect(interception.request.body).to.contain(
+        'Content-Disposition: form-data; name="uploadfile"; filename="uploadedFile.yml"',
+      );
+      expect(interception.request.body).to.contain('Content-Type: application/x-yaml');
+      expect(interception.request.body).to.contain('test-yaml-pipeline-content');
+
+      expect(interception.request.query).to.eql({
+        name: 'New pipeline',
+        description: 'New pipeline description',
+      });
+    });
+
+    cy.wait('@refreshPipelines').then((interception) => {
+      /* eslint-disable camelcase */
+      expect(interception.request.query).to.containSubset({ sort_by: 'created_at desc' });
+    });
+
+    // Verify the uploaded pipeline is in the table
+    pipelinesTable.getRowByName('New pipeline').find().should('exist');
+  }
+
+  importPipelineFromUrl(
+    projectName: string,
+    initialMockPipeline: PipelineKFv2,
+    createPipelineAndVersionParams: CreatePipelineAndVersionParams,
+  ) {
+    const createdMockPipeline = buildMockPipelineV2(createPipelineAndVersionParams.pipeline);
+
+    // Intercept upload/re-fetch of pipelines
+    pipelineImportModal
+      .mockCreatePipelineAndVersion(createPipelineAndVersionParams, projectName)
+      .as('createPipelineAndVersion');
+    pipelinesTable
+      .mockGetPipelines([initialMockPipeline, createdMockPipeline], projectName)
+      .as('refreshPipelines');
+    pipelinesTable.mockGetPipelineVersions(
+      [buildMockPipelineVersionV2(createPipelineAndVersionParams.pipeline_version)],
+      'new-pipeline',
+      projectName,
+    );
+
+    // Wait for the pipelines table to load
+    pipelinesTable.find().should('exist');
+
+    // Open the "Import pipeline" modal
+    pipelinesGlobal.findImportPipelineButton().click();
+
+    // Fill out the "Import pipeline" modal and submit
+    pipelineImportModal.shouldBeOpen();
+    pipelineImportModal.fillPipelineName('New pipeline');
+    pipelineImportModal.findImportPipelineRadio().check();
+    pipelineImportModal.findPipelineUrlInput().type('https://example.com/pipeline.yaml');
+    pipelineImportModal.submit();
+
+    // Wait for upload/fetch requests
+    cy.wait('@createPipelineAndVersion');
+    cy.wait('@refreshPipelines');
+
+    // Verify the uploaded pipeline is in the table
+    pipelinesTable.getRowByName('New pipeline').find().should('exist');
+  }
+
+  uploadPipelineVersion(
```

Comment:
I should have been more clear on my suggestion to move code to the page objects and also that not all test code should be moved to page objects.

We need to consider that page objects must be reusable for both mock and live cluster testing. Especially when using a modal, we should not assume that the modal is being invoked from a particular page.

There is value in abstracting some common interactions and assertions into page objects. But you should also consider creating shared test utilities that are specific to your mock tests to be reused across pages.

As part of the pipeline import modal, these functions have too much knowledge about the overall state of the app as well as are not fully parameterized to be reusable in live cluster testing for different scenarios.